packages/web3/test/e2e/get_uncle.test.ts

```diff
@@ -0,0 +1,88 @@
+/*
+This file is part of web3.js.
+
+web3.js is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+web3.js is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with web3.js.  If not, see <http://www.gnu.org/licenses/>.
+*/
+import { Numbers } from 'web3-types';
+
+import Web3 from '../../src';
+import { getSystemE2ETestProvider } from './e2e_utils';
+import { closeOpenConnection, getSystemTestBackend } from '../shared_fixtures/system_tests_utils';
+import { toAllVariants } from '../shared_fixtures/utils';
+
+describe(`${getSystemTestBackend()} tests - getUncle`, () => {
+	const provider = getSystemE2ETestProvider();
+	const blockData: {
+		earliest: 'earliest';
+		latest: 'latest';
+		pending: 'pending';
+		finalized: 'finalized';
+		safe: 'safe';
+		blockNumber: number;
+		blockHash: string;
+	} = {
+		earliest: 'earliest',
+		latest: 'latest',
+		pending: 'pending',
+		finalized: 'finalized',
+		safe: 'safe',
+		blockNumber: getSystemTestBackend() === 'sepolia' ? 3240768 : 17029884,
+		blockHash:
+			getSystemTestBackend() === 'sepolia'
+				? '0xe5e66eab79bf9236eface52c33ecdbad381069e533dc70e3f54e2f7727b5f6ca'
+				: '0x2850e4a813762b2de589fa5268eacb92572defaf9520608deb129699e504cab2',
+	};
+
+	let web3: Web3;
+
+	beforeAll(() => {
+		web3 = new Web3(provider);
+	});
+
+	afterAll(async () => {
+		await closeOpenConnection(web3);
+	});
+
+	it.each(
+		toAllVariants<{
+			block:
+				| 'earliest'
+				| 'latest'
+				| 'pending'
+				| 'finalized'
+				| 'safe'
+				| 'blockHash'
+				| 'blockNumber';
+			uncleIndex: Numbers;
+		}>({
+			block: [
+				'earliest',
+				'latest',
+				'pending',
+				'safe',
+				'finalized',
+				'blockHash',
+				'blockNumber',
+			],
+			uncleIndex: ['0x1', '1', 1, BigInt(1)],
+		}),
+	)('getUncle', async ({ block, uncleIndex }) => {
+		// TODO Returns Position Index is incorrect RPC error
+		// when the client is Nethermind, but not Geth
```

Comment:
Another provider discrepancy issue

Reply:
Could you post exact error? is it again some field rename issue?

Reply:
btw uncle blocks are no more in POS NW, but we should support it if some one is using Eth node on different consensus NW.

Reply:
@jdevcs This is the error received

```typescript
InvalidResponseError: Returned error: Position Index is incorrect
          at Web3RequestManager._processJsonRpcResponse (web3.js/packages/web3-core/src/web3_request_manager.ts:363:11)
          at Web3RequestManager.<anonymous> (web3.js/packages/web3-core/src/web3_request_manager.ts:207:16)
          at Generator.next (<anonymous>)
          at fulfilled (web3.js/packages/web3-core/lib/commonjs/web3_request_manager.js:21:58)
          at processTicksAndRejections (node:internal/process/task_queues:95:5) {
        innerError: { code: -32602, message: 'Position Index is incorrect' },
        code: 101,
        data: undefined,
        request: {
          jsonrpc: '2.0',
          id: '8f872b5e-9a66-486b-9c74-f2699167191a',
          method: 'eth_getUncleByBlockNumberAndIndex',
          params: [ '0x317340', '0x0' ]
        }
      }
```

I'm assuming the error is because there are no longer uncles, as you've mentioned, but Geth doesn't throw an error for the same request. I don't think we should do anything special for this circumstance as it's up to the provider whether they through an error for this or not, and web3.js is simply just passing along the error response to the user